src/storage.rs

```diff
@@ -75,10 +92,24 @@ pub trait Storage {
     /// the slice of entries returned will always have length at least 1 if entries are
     /// found in the range.
     ///
+    /// If the entries are fetched asynchorously, it would return LogTemporarilyUnavailable,
+    /// and application needs to call `on_entries_fetched(context)` to trigger re-fetch of the entries
+    /// after the storage finishes fetching the entries.   
+    ///
+    /// Constraint: only [low, high] covers unapplied entries are permitted to fetch entries
+    /// asynchorously, otherwise some callers may panic when meeting LogTemporarilyUnavailable.
```

Comment:
```suggestion
    /// Constraint: due to implementation limits, currently only when [low, high] covers any applied
    /// entries are permitted to fetch entries asynchorously, otherwise it may panic.
```